engine/flink/avro-util/src/main/scala/pl/touk/nussknacker/engine/avro/SchemaRegistryClient.scala

```diff
@@ -0,0 +1,25 @@
+package pl.touk.nussknacker.engine.avro
+
+import io.confluent.kafka.schemaregistry.client.SchemaMetadata
+import org.apache.avro.Schema
+import pl.touk.nussknacker.engine.kafka.KafkaConfig
+
+trait SchemaRegistryClient extends Serializable {
+
+  def getById(id: Int): Schema
+
+  def getSchemaMetadata(subject: String, version: Int): SchemaMetadata
+
+  def getLatestSchemaMetadata(name: String): SchemaMetadata
+
+  def schemaMetadata(subject: String, version: Option[Int]): SchemaMetadata =
+    version
+      .map(ver => getSchemaMetadata(subject, ver))
+      .getOrElse(getLatestSchemaMetadata(subject))
+}
+
+//It's only for internal providers usage..
+trait SchemaRegistryClientFactory[Client] extends Serializable {
+  type TypedSchemaRegistryClient = SchemaRegistryClient with Client
+  def createSchemaRegistryClient(kafkaConfig: KafkaConfig): TypedSchemaRegistryClient
```

Comment:
can it be `protected`?

Reply:
Yep

Reply:
Nope it's used at ConfluentSchemaRegistryProvider.createSchemaRegistryClient. But we can do protected this one from provieder as protected.

Reply:
ok, I've missed "provider" in "internal *providers* usage" :) should be "internal provider's usage" or "internal usage of provider"